timesketch/api/v1/resources.py

```diff
@@ -408,6 +467,291 @@ def post(self):
+                    line.setdefault('label', [])
+                    line['_id'] = event['_id']
+                    line['_type'] = event['_type']
+                    line['_index'] = event['_index']
+                    try:
+                        for label in line['timesketch_label']:
+                            if sketch.id != label['sketch_id']:
+                                continue
+                            line['label'].append(label['name'])
+                        del line['timesketch_label']
+                    except KeyError:
+                        pass
+
+                    lines.append(line)
+                data_frame = pd.DataFrame(lines)
+                del lines
+                data_frame.to_csv(fh, index=False)
+                fh.seek(0)
+                zip_file.writestr(
+                    'views/{0:s}.csv'.format(name), data=fh.read())
+
+                if not view.user:
+                    username = 'System'
+                else:
+                    username = view.user.username
+                meta = {
+                    'name': view.name,
+                    'view_id': view.id,
+                    'description': view.description,
+                    'query_string': view.query_string,
+                    'query_filter': view.query_filter,
+                    'query_dsl': view.query_dsl,
+                    'username': username,
+                    'sketch_id': view.sketch_id,
+                }
+                zip_file.writestr(
+                    'views/{0:s}.meta'.format(name), data=json.dumps(meta))
+
+            # TODO (kiddi): Add in aggregation group support.
+            # TODO (kiddi): Add in support for comments/stars/labels.
+            # TODO (kiddi): Add in support for all tagged events (includes
+            # a metadata file with a list of all tags and their count).
+
+        if is_archived:
+            _ = self._archive_sketch(sketch)
+
+        file_object.seek(0)
+        return send_file(
+            file_object, mimetype='zip',
+            attachment_filename='timesketch_export.zip')
+
+    def _unarchive_sketch(self, sketch):
+        """Unarchives a sketch by opening up all indices and removing labels.
+
+        Args:
+            sketch: Instance of timesketch.models.sketch.Sketch
+        """
+        if not sketch.has_label(ARCHIVE_LABEL):
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Unable to unarchive a sketch that wasn\'t already archived '
+                '(sketch: {0:d})'.format(sketch.id))
+
+        sketch.remove_label(ARCHIVE_LABEL)
+
+        indexes_to_open = []
+        for timeline in sketch.timelines:
+            timeline.remove_label(ARCHIVE_LABEL)
+            search_index = timeline.searchindex
+            search_index.remove_label(ARCHIVE_LABEL)
+            indexes_to_open.append(search_index.index_name)
+
+        self.datastore.client.indices.open(','.join(indexes_to_open))
```

Comment:
Let's add a TODO here about moving this to elastic.py datastore abstraction.
```suggestion
        # TODO: Move to lib/datastores/elastic.py
        self.datastore.client.indices.open(','.join(indexes_to_open))
```

Reply:
we can do that... there are quite a few other `client.indices.X` calls in the resources file, not sure if you want to move all of them.

Reply:
Yeah, let me take care of that later :)